app.py

```diff
@@ -69,3 +71,23 @@ def delete_event(title):
     global events
     events = list(filter(lambda x: x["title"] != title, events))
     return "", 204
+
+
+@app.route("/events/<string:title>/finish", methods=["PUT"])
+def finish_vote(title):
+    res = next(iter(filter(lambda x: x["title"] == title, events) or []), None)
+    if res:
+        res.update(is_finished=True)
+        return jsonify(res), 200
+    return jsonify({"Detail": "Couldn't finish voting. 'title' is required."}), 400
+
+
+@app.route("/events/<string:title>/vote", methods=["PUT"])
+def vote_event(title):
```

Comment:
The business logic of voting is to select one of the dates provided. We don't need a voting system for the event itself.